mno_analysis_tools/compute_window_of_downtime.py

```diff
@@ -0,0 +1,89 @@
+import json
+import pytz
+import argparse
+from datetime import datetime
+
+from temba_client.v2 import Message
+from core_data_modules.logging import Logger
+
+
+log = Logger(__name__)
+log.set_project_name("ComputeWindowOfDowntime")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Compute maximum window of time with 0 messages")
+    parser.add_argument("input_file", metavar="input file", type=argparse.FileType(mode="r"),
+        help="File to read the raw data downloaded as json.",
+    )
+    parser.add_argument("output_file", metavar="output file", type=argparse.FileType(mode="w"),
+        help="File to write the raw data downloaded as json.",
+    )
+    parser.add_argument("communication_medium", metavar="operator",
+        help="Operator that you'll need to analyze",
+    )
+    parser.add_argument("direction", metavar="direction of message",
+        help="Direction of the message either sent or received by the operator",
+    )
+    parser.add_argument("start_date", metavar="start date", 
+        type=lambda s: datetime.strptime(s, '%Y-%m-%d').astimezone(pytz.UTC),
+        help="The start date (yyyy-mm-dd) from which the window of downtime will be computed.",
+    )
+    parser.add_argument("end_date", metavar="end date",
+        type=lambda s: datetime.strptime(s, '%Y-%m-%d').astimezone(pytz.UTC),
+        help="The end date (yyy-mm-dd) to which the window of downtime computation will end",
+    )
+
+    args = parser.parse_args()
+
+    input_file = args.input_file
+    output_file = args.output_file
+    operator = args.communication_medium
+    msg_direction = args.direction
+    start_date = args.start_date
+    end_date = args.end_date
+
+    with input_file as file:
+        file_content = file.readline()
+        output = json.loads(file_content)
+        messages = [Message.deserialize(val) for val in output]
+        log.info(f"Read {len(messages)} messages...")
+
+    period_with_msg = []
+    generated_outputs = []
+
+    period_with_msg.insert(0, start_date)
+    for msg in messages:
+        communication_medium = msg.urn.split(":")[0]
+        direction = msg.direction
+        if communication_medium == operator and direction == msg_direction:
```

Comment:
When would direction == msg.direction ever be false?

Reply:
its direction == msg_direction  where 'msg_direction' is an argument

Reply:
I misread msg_direction as msg.direction. Having msg_direction represent something other than msg.direction is confusing, and the assignment of direction to be msg.direction didn't help.

How about changing this to:
`if operator == target_operator and msg.direction == target_direction`

or similar